add Smalltalk support

Nef10 opened the pull request:
currently `SqueakTrunk`, `Squeak5.0`, `Squeak4.6` and `Squeak4.5` images are supported

joshk commented:
Wow, fantastic!

@BanzaiMan could you please review this when you have time

fniephaus commented on lib/travis/build/script/smalltalk.rb:
```diff
@@ -0,0 +1,26 @@
+          sh.cmd 'sudo apt-get install --no-install-recommends libc6:i386 ' +
+                   'libuuid1:i386', retry: true
+          sh.if '-z "$SMALLTALK"' do
+            sh.export 'SMALLTALK', 'Squeak5.0'
```
This is not necessary anymore :smiley: (see hpi-swa/filetreeCI#6)

BanzaiMan commented on lib/travis/build/script/smalltalk.rb:
```diff
@@ -0,0 +1,26 @@
+      class Smalltalk < Script
+        DEFAULTS = {}
+
+        def setup
```
`setup` runs after `sudo` is disabled, so this won't work. You need to do everything you need with `sudo` in `configure`. See http://docs.travis-ci.com//user/languages/community-supported-languages/#Code

BanzaiMan commented on lib/travis/build/script/smalltalk.rb:
```diff
@@ -0,0 +1,26 @@
+          sh.cmd "cd $HOME"
+          sh.cmd "wget -q -O filetreeCI.zip https://github.com/hpi-swa/filetreeCI/archive/master.zip"
+          sh.cmd "unzip -q -o filetreeCI.zip"
+          sh.cmd "cd filetreeCI-*"
```
In general, I advise against use of `cd` to move around. If you need to change working directories, try `pushd`/`popd` in pairs so that our commands do not affect the user's working directory.

Nef10 commented:
@BanzaiMan we have fixed the issues and changed the smalltalk version matrix as described in the document you linked, related PR:
travis-ci/travis-web#404
travis-ci/travis-core#482

BanzaiMan commented:
Thank you! I've deployed this code to staging (without matrix changes to other components). Could you try this on https://staging.travis-ci.org?

fniephaus commented:
> The command "eval sudo apt-get install --no-install-recommends libc6:i386 libuuid1:i386" failed.

https://staging.travis-ci.org/fniephaus/SWT-Demo/builds/446506#L91

BanzaiMan commented:
Interesting. The package index on the image appears to be out of date. If I run `sudo apt-get update -qq` (`-qq` to suppress output) before, the packages are found and installed. Could you add that, please?

jchromik commented:
Done!

BanzaiMan commented:
New version is deployed to staging. Could you try again? You can either re-run the previously failing build, or push a new commit.

fniephaus commented:
Looks like it's working! :smiley: 

Is there any way to get right of this:

> The command "popd; popd" exited with 0.

https://staging.travis-ci.org/fniephaus/SWT-Demo/builds/446506#L201

BanzaiMan commented on lib/travis/build/script/smalltalk.rb:
```diff
@@ -0,0 +1,37 @@
+            sh.cmd "export FILETREE_CI_HOME=\"$(pwd)\""
+          end
+          sh.cmd "$FILETREE_CI_HOME/run.sh"
+          sh.cmd "popd; popd"
```
Add `, echo: false` to suppress output.

BanzaiMan commented on lib/travis/build/script/smalltalk.rb:
```diff
@@ -0,0 +1,37 @@
+        def setup
+          super
+          sh.cmd "export PROJECT_HOME=\"$(pwd)\""
+          sh.cmd "pushd $HOME"
```
Same here.

BanzaiMan commented on lib/travis/build/script/smalltalk.rb:
```diff
@@ -0,0 +1,37 @@
+            sh.echo 'Downloading and extracting filetreeCI', ansi: :yellow
+            sh.cmd "wget -q -O filetreeCI.zip https://github.com/hpi-swa/filetreeCI/archive/master.zip"
+            sh.cmd "unzip -q -o filetreeCI.zip"
+            sh.cmd "pushd filetreeCI-*"
```
…and here.

Nef10 commented:
so the master is not building with the matrix:
```
smalltalk:
  - SqueakTrunk
  - Squeak5.0
  - Squeak4.6
  - Squeak4.5
```
because travis-ci/travis-core#482 is not on staging?
all branches with only one version are building

BanzaiMan commented:
@Nef10 Affirmative. Matrix expansion is not happening at the moment.

BanzaiMan commented:
https://github.com/travis-ci/travis-build/commit/94b98afabe87cc14dd645f4e9456cd9ff67c5cdd is pushed to staging.

Nef10 commented:
@fniephaus seems like there are files missing for OS X:
https://www.hpi.uni-potsdam.de/hirschfeld/artefacts/filetreeci/vms/cog_linux_spur.min.tar.gz
https://www.hpi.uni-potsdam.de/hirschfeld/artefacts/filetreeci/vms/cog_osx.tar.gz

fniephaus commented:
@Nef10 I never uploaded them, because I never thought we could do OS X for Smalltalk on Travis.
Now things have changed quickly, so I'm happy to add them. In fact, they are already online :)

Nef10 commented:
@fniephaus thanks, working now
@BanzaiMan in the last commit I hopefully fixed that the test are now going to fail instead of error, everything else seems to work

BanzaiMan commented:
OK. It's deployed to staging.

BanzaiMan commented:
Deployed to production! Thanks!
